### DISCUSSION TITLE
Misleading environment variable name: APACHE_PORT

### DISCUSSION THREAD

**Discussion by TimKochDev** [nextcloud/all-in-one#5796] [Ideas] [answered]
I set up nextcloud-aio behind a reverse proxy. Therefore, I had to set `APACHE_PORT` to the port on the host system that I want to listen the nextcloud container on. While I am sure that `APACHE_PORT` is a technically correct name (I guess nextcloud-aio starts some kind of apache server?), this is not self-explanatory. IMHO, we should rename it to `NEXTCLOUD_PORT` to better represent the intent of the variable. I am sure I would have mastered the installation of the aio container behind a reverse proxy much faster if the documentation had stated something like:

> `NEXTCLOUD_PORT`: The port nextcloud should listen on after the AIO container sets up everything for you. Defaults to 80.

(I am not sure about 80 being the default port. This is just an example.)

**Comment by Zoey2936** [comment]
There is no "default" port if you don't set the env aio can't run behind a reverse proxy. And binds 443udp+tcp

**Reply by TimKochDev** [reply]
> There is no "default" port if you don't set the env aio can't run behind a reverse proxy. And binds 443udp+tcp

You mean APACHE_PORT as a _variable_ has no default, right? Because if I understand the nextcloud-aio configuration correctly, nextcloud has a default _behavior_ when APACHE_PORT is not set. And that is serving as if there was no reverse proxy (which to my knowledge means serving on the default HTTP ports, right?)

See, you as a collaborator know that Apache is used and what for. From that knowledge you can deduct what a variable named APACHE_PORT might be good for. For me, just wanting to install this great project within a few minutes on my home server which is already running other services behind a reverse proxy, for me APACHE_PORT doesn't make sense. Do I have to install Apache myself? Is it a problem that I use Traefik instead of Apache? I don't know. I first had to dig into the documentation to learn that APACHE_PORT simply is the port I want nextcloud to listen on.

This is just the humble feedback of a potential nextcloud user who initially got confused and scared off simply by the naming of a variable ;)

**Reply by Zoey2936** [reply, marked as answer]
yes I understand what you mean. If nothing is set then the mastercontainer redirects 80 to 443 and inside the apache container caddy listens on 443 (and caddy makes tls and proxies to other services like nextcloud (handeld by apache in the same containter, while php runs in the nextcloud container) and also caddy proxies to other containers like collabora). When you set apache port, then caddy does the sam but runs on http on your selected port and trusts other proxies

**Comment by TimKochDev** [comment]
FYI I just [published a medium story](https://medium.com/@timkochdev/running-nextcloud-aio-alongside-coolify-a-practical-guide-8b0de174ba76#:~:text=Note%3A%20Don%E2%80%99t%20let%20the%20APACHE_PORT%20name%20confuse%20you%20%E2%80%94%20it%E2%80%99s%20simply%20defining%20the%20port%20where%20Nextcloud%20should%20listen%2C%20not%20specifically%20related%20to%20Apache.) about how I installed nextcloud on my homeserver alongside Coolify (and thus Traefik):
